### DISCUSSION TITLE
Problem with laser power

### DISCUSSION THREAD

**Discussion by mgardenal** [arkypita/LaserGRBL#1839] [Q&A]
I'm finishing converting a Chinese machine using cnc Shield + Grbl
And I'm doing tests to get close to what the original software of the machine did, but it seems that the laser power does not obey the configuration..
It is configured $30 1000, but even using 250, 300 of power, it burns a lot, I'll send an example of how it was with the software and original board.

**Is there any configuration that I can get to this result?
In the original software there was a setting of
depth of carving and he would sink without turning the piece black.**
![1654628946681](https://user-images.githubusercontent.com/106627278/172463092-53f676af-458b-4d44-932b-a6410386c87d.jpg)

![1654628946659](https://user-images.githubusercontent.com/106627278/172463114-74a006db-bb59-4fcb-ab35-43df5815beb1.jpg)

**This is an example of how it looks on LaserGrbl**
![1654628946637](https://user-images.githubusercontent.com/106627278/172463104-4352106d-ac7a-4561-9bc3-0f4046d31fc4.jpg)

**Comment by arkypita** [comment]
It is not only the power that affects the result but it is also the speed. What speed have you set? Are you sure that the machine is going at the target speed or maybe it is going slower due to constraints given by speed/accelerations limits? Are you using M3 or M4? 

Is the laser driver able to handle grbl default PWM frequency (1KHz)? Usually yes, but the old controller board may use a different frequency.

In the last picture laser is totally out of focus.

Suggested reading: https://lasergrbl.com/wp-content/uploads/2020/05/Grbl-Configuration-ENG.pdf

**Reply by arkypita** [reply]
Also please confirm you have grbl v1.1

**Reply by mgardenal** [reply]
Yes I use version 1.1

**Reply by mgardenal** [reply]
M3 use
The laser I do not have this information

**Comment by mgardenal** [comment]
Please help me 😅 the machine keeps stopping out of nowhere, this message 0 appears in the program

![IMG_20220607_212438](https://user-images.githubusercontent.com/106627278/172743062-3b1c8f61-6b06-4342-abe3-ec711ec5f749.jpg)

**Reply by arkypita** [reply]
"o" (not zero) is probably the message "ok" sent by the machine to acknowledge the last sent command (G0 X36 Y65 S0) but missing the "k" character, so LaserGRBL is not able to recognize it and still waiting for the reply. 

The cause of this issue is 99% electrical noise.

Please read https://lasergrbl.com/faq carefully, it contains most common cause and solution troubleshooting.

Please note the red buffer indicator, showing that some data (the last command) is in queue from too much time. The small "thunder" icon near buffer indicator allow you to "force" LaserGRBL ignoring the missing acknowledge. Try to click it and see what's going on.

**Reply by mgardenal** [reply]
I'll check the electrical. 
Can you help me with the issue of the recording depth?  to get close to the quality of the first images?

**Reply by mgardenal** [reply]
> "o" (não zero) é provavelmente a mensagem "ok" enviada pela máquina para reconhecer o último comando enviado (G0 X36 Y65 S0), mas falta o caractere "k", então o LaserGRBL não é capaz de reconhecê-lo e ainda está aguardando a resposta.
> 
> A causa desse problema é 99% de ruído elétrico.
> 
> Por favor, leia https://lasergrbl.com/faq com cuidado, ele contém a causa mais comum e solução de problemas de solução.
> 
> Observe o indicador de buffer vermelho, mostrando que alguns dados (o último comando) estão na fila por muito tempo. O pequeno ícone "trovão" próximo ao indicador de buffer permite que você "força" o LaserGRBL ignorando a confirmação ausente. Tente clicar nele e ver o que está acontecendo.

this is the information that appears when you click on the "thunder"
I don't know what the communication error could be, I've already tested it with other usb cables, I've already reviewed the power cables, motor..
Please help me :(

**Reply by mgardenal** [reply]
![WhatsApp Image 2022-06-14 at 22 12 51](https://user-images.githubusercontent.com/106627278/173715754-dc8f4477-047b-4652-a04b-7d84bf4b1847.jpeg)

**Reply by arkypita** [reply]
> Please help me :(

There is nothing I can do here to help you.
If LaserGRBL does not receive back the "ok" acknowledge message it cannot send next command.

LaserGRBL is able to detect and self recover this situation by reading the buffer size from the machine and compare it with its knowledge.
If the machine say that the buffer is free but lasergrbl think it's full, LaserGRBL do a self-reset of the buffer count.

This feature can be enabled by telling the machine to report the buffer size in it's status message.
This is done by setting $10=3

![image](https://user-images.githubusercontent.com/8782035/173774541-2d0382c6-49cd-4f6a-8e92-6db181f5c09b.png)
